luaui/Widgets/gui_gridmenu.lua

```diff
@@ -2081,43 +2249,43 @@ local function handleButtonHover()
 			if currentCategory or labBuildModeActive then
 				-- back button
 				if backRect and backRect:contains(x, y) then
-					hoveredButton = backRect:getId()
+					setHoveredRect(backRect)
 					hoveredButtonNotFound = false
-					if WG['tooltip'] then
-						local text = "\255\240\240\240" .. Spring.I18N('ui.buildMenu.homePage')
-						WG['tooltip'].ShowTooltip('buildmenu', text)
+					if WG["tooltip"] then
+						local text = "\255\240\240\240" .. Spring.I18N("ui.buildMenu.homePage")
+						WG["tooltip"].ShowTooltip("buildmenu", text)
 					end
```

Comment:
This snippet of code is repeated many many times.
I would try moving it to a locally scoped function just for the sake of code organization and making maintenance easier. Performance hit should be negligible

at the top of the function after "local hoveredButtonNotFound = false"

something along these lines?
```lua
local function setHoveredRectWithTooltip(rect, tooltipTranslationKey)
    setHoveredRect(rect)
    hoveredButtonNotFound = false
    if WG["tooltip"] then
         local text = "\255\240\240\240" .. Spring.I18N(tooltipTranslationKey)
         WG["tooltip"].ShowTooltip("buildmenu", text)
    end
end
```

the if check could also potentially be moved into that function, but the code might get a bit too messy that way

Reply:
I am purposefully avoiding logic inside handle button hover. I want to focus on stability before (if I do) tackling the mess that is this function for now.

Reply:
fair enough

Reply:
Done, good call!